homeassistant/components/device_tracker/tile.py

```diff
@@ -34,89 +32,106 @@
-            type_whitelist=self._types, show_inactive=self._show_inactive)
-
-        if not self.devices:
-            _LOGGER.warning('No Tiles found')
-            return
-
-        for dev in self.devices:
-            dev_id = 'tile_{0}'.format(slugify(dev['name']))
-            lat = dev['tileState']['latitude']
-            lon = dev['tileState']['longitude']
-
-            attrs = {
-                ATTR_ALTITUDE: dev['tileState']['altitude'],
-                ATTR_CONNECTION_STATE: dev['tileState']['connection_state'],
-                ATTR_IS_DEAD: dev['is_dead'],
-                ATTR_IS_LOST: dev['tileState']['is_lost'],
-                ATTR_RING_STATE: dev['tileState']['ring_state'],
-                ATTR_VOIP_STATE: dev['tileState']['voip_state'],
-            }
-
-            self.see(
-                dev_id=dev_id,
-                gps=(lat, lon),
-                attributes=attrs,
-                icon=DEFAULT_ICON
-            )
+        self._async_see = async_see
+        self._client = client
+        self._hass = hass
+        self._show_inactive = show_inactive
+        self._types = types
+
+    async def async_init(self):
+        """Further initialize connection to the Tile servers."""
+        from pytile.errors import TileError
+
+        try:
+            await self._client.get_session()
+            await self._async_update()
+
+            async_track_time_interval(
+                self._hass, self._async_update, DEFAULT_SCAN_INTERVAL)
+
+            return True
+        except TileError as err:
+            _LOGGER.error('Unable to set up Tile scanner: %s', err)
+            return False
+
+    async def _async_update(self, now=None):
+        """Update info from Tile."""
+        from pytile.errors import SessionExpiredError, TileError
+        _LOGGER.debug('Updating Tile data')
+
+        try:
+            tiles = await self._client.tiles.all(
+                whitelist=self._types, show_inactive=self._show_inactive)
+
+            if not tiles:
+                _LOGGER.warning('No Tiles found')
+                return
+
+            for tile in tiles:
+                await self._async_see(
+                    dev_id='tile_{0}'.format(slugify(tile['name'])),
+                    gps=(
+                        tile['tileState']['latitude'],
+                        tile['tileState']['longitude']),
+                    attributes={
+                        ATTR_ALTITUDE: tile['tileState']['altitude'],
+                        ATTR_CONNECTION_STATE:
+                            tile['tileState']['connection_state'],
+                        ATTR_IS_DEAD: tile['is_dead'],
+                        ATTR_IS_LOST: tile['tileState']['is_lost'],
+                        ATTR_RING_STATE: tile['tileState']['ring_state'],
+                        ATTR_VOIP_STATE: tile['tileState']['voip_state'],
+                    },
+                    icon=DEFAULT_ICON)
+        except SessionExpiredError:
```

Comment:
Only wrap the call that you expect to error with try... except. I think that's `await self._client.tiles.all` in this case.